ClassTrigger.trigger.txt

```diff
@@ -0,0 +1,23 @@
+trigger ClassTrigger on Class__c (before insert, before delete, before update){
+    
+    // trigger code to restrict class deletion if class has more than one female students
+    if(trigger.isDelete){
+        List<AggregateResult> students = [SELECT Class__r.id, COUNT(Name) FROM Student__c WHERE Sex__c = 'female' AND Class__c IN:  Trigger.old GROUP BY Class__r.id HAVING COUNT(Name) > 1];
+        for (AggregateResult everyStudent : students){
+            Trigger.oldMap.get((id)everyStudent.get('Class__r.id')).addError('This class cannot be deleted as it contains more than one female students');    

```

Comment:
We can use constants for error messages so that in case we need to use a single error message at multiple places it is easy to track and change